nativelib/src/main/scala/scala/scalanative/misc/Scope.scala

```diff
@@ -0,0 +1,63 @@
+package scala.scalanative.misc
+
+/** Scoped implicit lifetime.
+ *
+ *  The main idea behind the Scope is to encode resource lifetimes through
+ *  a concept of an implicit scope. Scopes are necessary to acquire resources.
+ *  They are responsible for disposal of the resources once the evaluation exits
+ *  the demarkated block in the source code.
+ *
+ *  See https://www.youtube.com/watch?v=MV2eJkwarT4 for details.
+ */
+trait Resource {
+  def close(): Unit
+}
+
+@annotation.implicitNotFound(msg = "Resource acquisition requires a scope.")
+trait Scope {
+
+  /** Push resource onto the resource stack. */
+  def acquire[R <: Resource](res: R): R
+
+  /** Clean up all the resources in FIFO order. */
+  def close(): Unit
+}
```

Comment:
I mean a separate subproject in our sbt build inside of the main repo. It's perfectly fine to move stuff around as part of a PR. :)